openedx/core/djangoapps/oauth_dispatch/management/commands/generate_jwt_signing_key.py

```diff
@@ -123,15 +123,23 @@ def _generate_key_id(self, size, chars=string.ascii_uppercase + string.digits):
     def _generate_key_pair(self, key_size, key_id):
         log.info('Generating new JWT signing keypair for key id %s.', key_id)
         rsa_key = RSA.generate(key_size)
-        rsa_jwk = jwk.RSAKey(kid=key_id, key=rsa_key)
-        return rsa_jwk
+        algo = get_default_algorithms()['RS512']
+        pem = rsa_key.export_key('PEM').decode()
+        rsa_jwk = json.loads(algo.to_jwk(algo.prepare_key(pem)))
+        public_rsa_jwk = json.loads(algo.to_jwk(algo.prepare_key(pem).public_key()))
```

Comment:
Suggestion on tightening up this code a little:
```suggestion
        algo = get_default_algorithms()['RS512']
        key_data = algo.prepare_key(rsa_key.export_key('PEM').decode())
        rsa_jwk = json.loads(algo.to_jwk(key_data))
        public_rsa_jwk = json.loads(algo.to_jwk(key_data.public_key()))
```